ratis-common/src/main/java/org/apache/ratis/protocol/DataStreamPacket.java

```diff
@@ -23,6 +23,8 @@
 public interface DataStreamPacket {
   long getStreamId();
 
+  RaftGroupId getGroupId();
```

Comment:
We only have to pass groupId in the RaftClientRequest (header) but not every packet.

Reply:
I see. so the header indicates which group statement to use for a stream id. 

At this moment, for a stream (with a stream id), it has a header(that has a RaftClientRequest with groupId), which indicates a group  on a RaftServer.